Description: Fix compatibility of LDAP support with Erlang R16
 The asn1rt_ber_bin has been removed from Erlang/OTP R16A,
 so this patch removes calls to these functions since they are
 only used to do sanity checks which appear to be redundant.
Author: Badlop <badlop@process-one.net>
Bug: https://support.process-one.net/browse/EJAB-1612
Forwarded: not-needed
Applied-Upstream: 2.1.12, 2704378d43035474c5f3d8a656b81c7bc28b5ff9
Last-Update: 2013-10-24
---
This patch header follows DEP-3: http://dep.debian.net/deps/dep3/
--- a/src/eldap/eldap.erl
+++ b/src/eldap/eldap.erl
@@ -760,7 +760,6 @@
 %%  {'EXIT', Reason} - Broke
 %%-----------------------------------------------------------------------
 recvd_packet(Pkt, S) ->
-    check_tag(Pkt),
     case asn1rt:decode('ELDAPv3', 'LDAPMessage', Pkt) of
 	{ok,Msg} ->
 	    Op = Msg#'LDAPMessage'.protocolOp,
@@ -873,7 +872,6 @@
 %%  {'EXIT', Reason} - Broken packet
 %%-----------------------------------------------------------------------
 recvd_wait_bind_response(Pkt, S) ->
-    check_tag(Pkt),
     case asn1rt:decode('ELDAPv3', 'LDAPMessage', Pkt) of
 	{ok,Msg} ->
 	    ?DEBUG("~p", [Msg]),
@@ -906,12 +904,6 @@
     end.
 
 
-%%% Sanity check of received packet
-check_tag(Data) ->
-    {_Tag, Data1, _Rb} = asn1rt_ber_bin:decode_tag(Data),
-    {{_Len,_Data2}, _Rb2} = asn1rt_ber_bin:decode_length(Data1),
-    ok.
-
 close_and_retry(S, Timeout) ->
     catch (S#eldap.sockmod):close(S#eldap.fd),
     Queue = dict:fold(
